network/builder/src/builder.rs

```diff
@@ -295,58 +239,13 @@ impl NetworkBuilder {
         self.conn_mgr_reqs_tx.clone()
     }
 
-    fn supported_protocols(&self) -> SupportedProtocols {
-        self.direct_send_protocols
-            .iter()
-            .chain(&self.rpc_protocols)
-            .into()
-    }
-
-    /// Add a handler for given protocols using raw bytes.
-    fn inner_add_protocol_handler(
-        &mut self,
-        rpc_protocols: Vec<ProtocolId>,
-        direct_send_protocols: Vec<ProtocolId>,
-        queue_preference: QueueStyle,
-        max_queue_size_per_peer: usize,
-        counter: Option<&'static IntCounterVec>,
-    ) -> (
-        PeerManagerRequestSender,
-        libra_channel::Receiver<(PeerId, ProtocolId), PeerManagerNotification>,
-        ConnectionRequestSender,
-        conn_notifs_channel::Receiver,
-    ) {
-        self.direct_send_protocols
-            .extend(direct_send_protocols.clone());
-        self.rpc_protocols.extend(rpc_protocols.clone());
-        let (network_notifs_tx, network_notifs_rx) = libra_channel::new(
-            queue_preference,
-            NonZeroUsize::new(max_queue_size_per_peer).unwrap(),
-            counter,
-        );
-        for protocol in rpc_protocols
-            .iter()
-            .chain(direct_send_protocols.iter())
-            .cloned()
-        {
-            self.upstream_handlers
-                .insert(protocol, network_notifs_tx.clone());
-        }
-        let (connection_notifs_tx, connection_notifs_rx) = conn_notifs_channel::new();
-        // Auto-subscribe all application level handlers to connection events.
-        self.connection_event_handlers.push(connection_notifs_tx);
-        (
-            PeerManagerRequestSender::new(self.pm_reqs_tx.clone()),
-            network_notifs_rx,
-            ConnectionRequestSender::new(self.connection_reqs_tx.clone()),
-            connection_notifs_rx,
-        )
-    }
-
     pub fn add_connection_event_listener(&mut self) -> conn_notifs_channel::Receiver {
-        let (tx, rx) = conn_notifs_channel::new();
-        self.connection_event_handlers.push(tx);
-        rx
+        match &mut self.peer_manager_builder {
+            Some(builder) => builder.add_connection_event_listener(),
```

Comment:
peer manager isn't optional. also I think it is better to use .as_ref().expect() instead.